### PR TITLE
Allow GET method on /auth with BasicAuth

### PR DESCRIPTION
**Pull request by zeehio**
## Proposed change

nginx has a convenient feature named [Authentication Based on Subrequest Result](https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-subrequest-authentication/) which allows to grant/deny access using basic authentication based on the result of a subrequest. That nginx module however only works with GET requests.

This Pull Requests allows users to provide auth credentials using Basic Auth in a GET request. This is convenient because it is then possible to use nginx to authenticate a third party app. Within the nginx server directive an add-on developer could use something like:

```
location = /.ha-auth {
    internal;
    proxy_pass              http://supervisor/auth;
    proxy_pass_request_body off;
    proxy_set_header        Content-Length "";
    proxy_set_header        X-Supervisor-Token "${SUPERVISOR_TOKEN}"; # The ${SUPERVISOR_TOKEN} needs to be set
}

location / {
    auth_request     /.ha-auth;
    auth_request_set $auth_status $upstream_status;
    proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header  Host $http_host;
    #proxy_set_header X-Remote-User $remote_user;
    #proxy_set_header   X-Forwarded-User $http_authorization;
    proxy_pass http://127.0.0.1:5000/;  # your addon could use the X-Remote-User header with the username
}
```

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/767

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [X] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by zeehio** [comment]
I still need to find time for testing, but it's ready for review

File: supervisor/api/auth.py [hunk 1]
```diff
@@ -67,11 +67,17 @@ async def auth(self, request: web.Request) -> bool:
...
+                raise HTTPBadRequest(text="json auth requires HTTP POST instead of GET")
             data = await request.json()
             return await self._process_dict(request, addon, data)
 
         # URL encoded
         if request.headers.get(CONTENT_TYPE) == CONTENT_TYPE_URL:
+            if request.method == "GET":
+                raise HTTPBadRequest(
+                    text="URL encoded auth requires HTTP POST instead of GET"
+                )
```
Comment (pvizeli) [inline supervisor/api/auth.py:80]: Of curse, that is correct, but do we need to guide people there? We can still just accept GET without future validation if GET or POST.

Comment (zeehio) [reply supervisor/api/auth.py]: I removed those checks. It becomes a one line pull request :-)

**Review by pvizeli — APPROVED** [review]

**Comment by frenck** [comment]
Really nice addition! Thanks, @zeehio 👍

**Pull request merged**